doc/whatsnew/fragments/7882.false_positive

```diff
@@ -0,0 +1,4 @@
+Fix a false positive for ``used-before-assignment`` after a variable
+annotated with a name guarded by ``if TYPE_CHECKING:``.
```

Comment:
```suggestion
Fix a false positive for ``used-before-assignment`` when a variable
is annotated with a name guarded by ``if TYPE_CHECKING:``.
```

Reply:
Unfortunately, that version sounds more like #7609, which was already solved. This bug doesn't happen until _after_ the variable has already been annotated. I wasn't sure how best to describe it succinctly. Open to suggestions.

Reply:
Aha I hear you. How about this or something similar?
```suggestion
Fix a false positive for ``used-before-assignment`` when a name guarded by ``if TYPE_CHECKING:`` is used as a type annotation in a function body and later re-imported in the same scope.
```